app/api/hello/atg.ts

```diff
@@ -0,0 +1,337 @@
+import { gql, useSubscription } from "@apollo/client";
+import { useState } from "react";
+
+// submitCodeSnippet.ts
+export const submitCodeSnippet = async ({
+  language,
+  schema,
+  code,
+}: {
+  language: string;
+  schema: string;
+  code: string;
+}) => {
+  try {
+    const headers = {
+      "Content-Type": "application/json",
+    };
+
+    const requestBody = {
```

Comment:
why aren't you using useMutation hook in apollo graphql instead of fetch ?

Reply:
Because from the apollo client what i wanted was the useSubscription hook only. Rest Api's can be just simply fetched using fetch as it is just a post request. Should i convert them to using useMutation hook?

Reply:
Apollo's useMutation hook is designed specifically for executing GraphQL mutations. By using it, you keep all your GraphQL-related logic consistent, whether it's queries, mutations, or subscriptions.You already setup graphql client so why are you not using it for mutations?